include/boost/math/special_functions/interval_sieve.hpp

```diff
@@ -0,0 +1,298 @@
+// Copyright 2020 Matt Borland and Jonathan Sorenson
+//
+// Use, modification and distribution are subject to the
+// Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt
+// or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_MATH_SPECIAL_FUNCTIONS_INTERVAL_SIEVE_HPP
+#define BOOST_MATH_SPECIAL_FUNCTIONS_INTERVAL_SIEVE_HPP
+
+#include <boost/dynamic_bitset.hpp>
+#include <boost/multiprecision/cpp_int.hpp>
+#include <boost/math/special_functions/prime_wheel.hpp>
+#include <cmath>
+#include <memory>
+#include <array>
+#include <cstdint>
+#include <bitset>
+
+namespace boost::math::detail
+{
+// boost::multiprecision and POD
+template<class Integer>
+inline double get_double(const Integer &x) noexcept
+{
+    return static_cast<double>(x);
+}
+
+template<class Integer, class PrimeContainer, class Container>
```

Comment:
So you understand the difference, what you have called `PrimeContainer` is _not_ actually a container. Or more precisely, the concrete type that instantiates here does not have to model all the interface of the _Container_ concept.
`primes_` just has to model a `RandomAccessRange`, i.e. it has `begin()`, `end()`, `size()` and `T& operator[](std::size_t i)`. A container can model that, but a pair of iterators, or an iterator-size pair can too.
So, to be precise, that template parameter is not a (Prime)Container, it's a (Prime)RandomAccessRange.